### PR TITLE
[SPARK-33575][SQL] Fix misleading exception for "ANALYZE TABLE ... FOR COLUMNS" on temporary views

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

This PR proposes to fix the exception message for `ANALYZE TABLE ... FOR COLUMNS` on temporary views. 

The current behavior throws `NoSuchTableException` even if the temporary view exists:
```
sql("CREATE TEMP VIEW t AS SELECT 1 AS id")
sql("ANALYZE TABLE t COMPUTE STATISTICS FOR COLUMNS id")
org.apache.spark.sql.catalyst.analysis.NoSuchTableException: Table or view 't' not found in database 'db';
  at org.apache.spark.sql.execution.command.AnalyzeColumnCommand.analyzeColumnInTempView(AnalyzeColumnCommand.scala:76)
  at org.apache.spark.sql.execution.command.AnalyzeColumnCommand.run(AnalyzeColumnCommand.scala:54)
```

After this PR, more reasonable exception is thrown:
```
org.apache.spark.sql.AnalysisException: Temporary view `testView` is not cached for analyzing columns.;
[info]   at org.apache.spark.sql.execution.command.AnalyzeColumnCommand.analyzeColumnInTempView(AnalyzeColumnCommand.scala:74)
[info]   at org.apache.spark.sql.execution.command.AnalyzeColumnCommand.run(AnalyzeColumnCommand.scala:54)
```

### Why are the changes needed?

To fix a misleading exception.

### Does this PR introduce _any_ user-facing change?

Yes, the exception thrown is changed as shown above.

### How was this patch tested?

Updated existing test.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala [hunk 1]
```diff
@@ -71,9 +71,8 @@ case class AnalyzeColumnCommand(
 
   private def analyzeColumnInTempView(plan: LogicalPlan, sparkSession: SparkSession): Unit = {
```
Comment (imback82) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala:72]: Note that `analyzeColumnInTempView` is called only if temp view (including global) exists: https://github.com/apache/spark/blob/d082ad0abfe0bc26760626ae0ecb415a8d508a1f/sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala#L45-L57

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala [hunk 2]
```diff
@@ -71,9 +71,8 @@ case class AnalyzeColumnCommand(
...
     if (!analyzeColumnInCachedData(plan, sparkSession)) {
-      val catalog = sparkSession.sessionState.catalog
-      val db = tableIdent.database.getOrElse(catalog.getCurrentDatabase)
-      throw new NoSuchTableException(db = db, table = tableIdent.identifier)
+      throw new AnalysisException(
+        s"Temporary view $tableIdent is not cached for analyzing columns.")
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala:75]: This message looks clearer.

**Comment by maropu** [comment]
Looks fine if the tests pass.

**Comment by imback82** [comment]
Thanks @maropu for taking a look!

cc @cloud-fan

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**